pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -168,9 +213,26 @@ func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl
 		// the desired state on the cluster
 		return ctrl.Result{Requeue: true}, nil
 	}
+
 	return ctrl.Result{}, nil
 }
 
+// finalize{{ .Resource.Kind }} will perform the required operations before delete the CR.
+func (r *{{ .Resource.Kind }}Reconciler) finalize{{ .Resource.Kind }}(log logr.Logger, cr *{{ .Resource.ImportAlias }}.{{ .Resource.Kind }}) error {
+	// TODO(user): Add the cleanup steps that the operator
+	// needs to do before the CR can be deleted. Examples
+	// of finalizers include performing backups and deleting
+	// resources that are not owned by this CR, like a PVC.
+	// The following implementation will raise an event
+	r.recorder.Event(cr, "Warning", "Deleting",
+		fmt.Sprintf("Custom Resource %s is being deleted from the namespace %s",
+		cr.Name,
+		cr.Namespace))
+
+	log.Info("Successfully finalized {{ lower .Resource.Kind }}")
+	return nil
+}
```

Comment:
```suggestion
func (r *{{ .Resource.Kind }}Reconciler) finalize{{ .Resource.Kind }}(cr *{{ .Resource.ImportAlias }}.{{ .Resource.Kind }}) {
	// TODO(user): Add the cleanup steps that the operator
	// needs to do before the CR can be deleted. Examples
	// of finalizers include performing backups and deleting
	// resources that are not owned by this CR, like a PVC.
	// The following implementation will raise an event
	r.recorder.Event(cr, "Warning", "Deleting",
		fmt.Sprintf("Custom Resource %s is being deleted from the namespace %s",
		cr.Name,
		cr.Namespace))
}
```
Nothing here can return an error so we need to clean up it.